embedchain/bots/slack.py

```diff
@@ -0,0 +1,93 @@
+import argparse
+import logging
+import os
+import signal
+import sys
+
+from dotenv import load_dotenv
+from flask import Flask, request
+from slack_sdk import WebClient
+
+from embedchain import App
+
+from .base import BaseBot
+
+load_dotenv()
+
+SLACK_BOT_TOKEN = os.environ.get("SLACK_BOT_TOKEN")
+
+
+class SlackBot(BaseBot):
+    def __init__(self):
+        self.client = WebClient(token=SLACK_BOT_TOKEN)
+        self.chat_bot = App()
+        self.recent_message = {"ts": 0, "channel": ""}
+        super().__init__()
+
+    def handle_message(self, event_data):
+        message = event_data.get("event")
+        if message and "text" in message and message.get("subtype") != "bot_message":
+            text: str = message["text"]
+            if float(message.get("ts")) > float(self.recent_message["ts"]):
+                self.recent_message["ts"] = message["ts"]
+                self.recent_message["channel"] = message["channel"]
+                if text.startswith("query"):
+                    _, question = text.split(" ", 1)
+                    try:
+                        response = self.chat_bot.chat(question)
+                        self.send_slack_message(message["channel"], response)
+                        print("Query answered successfully!")
```

Comment:
Can we please use logging.info everywhere?

Reply:
fixed. used `logging.error` for errors though.